# Day 5 — Matplotlib & Week 1 Mini-Notebook 

In [ ]:
# import libraries
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

task1: Load and clean a provided dataset with Pandas (handling any missing values)

In [ ]:
# read the dataset
df = pd.read_csv('Mobile Price/train.csv')

In [ ]:
print(f'null values: {df.isnull().sum()}')

null values: battery_power    2
blue             0
clock_speed      0
dual_sim         0
fc               0
four_g           0
int_memory       1
m_dep            1
mobile_wt        1
n_cores          0
pc               0
px_height        1
px_width         1
ram              1
sc_h             0
sc_w             0
talk_time        0
three_g          0
touch_screen     0
wifi             0
price_range      0
dtype: int64


In [ ]:
df.dropna(inplace=True) 

this dataset has no null values, so I manually deleted a small number of values, and deleting them does not affect the dataset.

task2:  Use NumPy to compute at least one derived numeric feature or summary statistic.

In [ ]:
df.describe()

Number of devices with touch screen: 1003


Describe

task4: Use groupby to compute an aggregate statistic per category and interpret the result in a 
Markdown cell.

In [ ]:
df.groupby("touch_screen")["ram"].mean() # I grouped the dataset based on devices that have a touch screen and calculated the average RAM for each group.

touch_screen
0    2157.753535
1    2091.592223
Name: ram, dtype: float64

## Average RAM Based on Touch Screen Feature

The groupby analysis shows the average RAM capacity for phones with and without touch screens.

Phones without a touch screen have a slightly higher average RAM (around 2158 MB) compared to phones with a touch screen (around 2092 MB).

This indicates that having a touch screen does not significantly increase RAM capacity in this dataset, and there is no strong relationship between the touch screen feature and RAM size.